### PR TITLE
trace-agent: remove deprecated apm_config.log_throttling

### PR DESCRIPTION
**Pull request by ahmed-mez**
### What does this PR do?

APMSP-945

Remove deprecated `apm_config.log_throttling`

### Motivation

The parameter was unused

### Describe how to test/QA your changes

Unit tests

### CODE REVIEW

**Comment by ahmed-mez** [comment]
Nice catch @knusbaum, thanks! Updated the PR.

**Review by maycmlee — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1536,12 +1536,6 @@ api_key:
   #
   # log_file: <APM_LOG_FILE_PATH>
 
-  ## @param log_throttling - boolean - default: true
-  ## @env DD_APM_LOG_THROTTLING - boolean - default: true
-  ## Limits the total number of warnings and errors to 10 for every 10 second interval.
-  #
-  # log_throttling: true
```
Comment (pgimalac) [inline pkg/config/config_template.yaml:1543]: I think if you remove a publicly documented setting you should have a release note to explain it

**Review by pgimalac — APPROVED** [review]
LGTM !

FYI there is one remaining use in `pkg/config/legacy/converter.go`

**Pull request merged**